staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go

```diff
@@ -131,6 +131,7 @@ type CreateSecretOptions struct {
 // NewSecretOptions creates a new *CreateSecretOptions with default value
 func NewSecretOptions(ioStreams genericiooptions.IOStreams) *CreateSecretOptions {
 	return &CreateSecretOptions{
+		Type: "Opaque",
```

Comment:
This field is optional and can be empty. You are changing the default to `Opaque` which breaks the backward compatibility.

/hold

Reply:
I don't know if there is some other way this might break backward compatibility, but it looks like an empty type is converted to Opaque as the default, just not by kubectl:

https://github.com/kubernetes/kubernetes/blob/a6ea7b8218d01b504fda2ef4988627fe4194fcaa/pkg/apis/core/types.go#L6025-L6026

https://github.com/kubernetes/kubernetes/blob/a6ea7b8218d01b504fda2ef4988627fe4194fcaa/pkg/apis/core/v1/defaults.go#L283-L287

When I create a secret without specifying `--type` it creates it as Opaque.

```
~ $ kubectl create secret generic my-secret --from-literal=foo=bar
secret/my-secret created
~ $ k describe secret my-secret
Name:         my-secret
Namespace:    default
Labels:       <none>
Annotations:  <none>

Type:  Opaque

Data
====
foo:  3 bytes
```

Reply:
Interesting. I didn't know that. But in any case, we can't set this explicitly in kubectl because when the default is changed in API level (even if this won't happen), kubectl value will be obsolete.

Reply:
Hi @mpuckett159 , would like your opinions here as well.
this change was inspired by the comment in another PR: https://github.com/kubernetes/kubernetes/pull/120337#discussion_r1408525014

Reply:
can we instead just add it as a text in the help message. the real target is to mention that there is a default value to the option. However in the discussion mentioned earlier @mpuckett159 mentioned it is better to add it as automatic value to come up when we use `kubectl create secret  generic --help`
WDYT?

Reply:
@brianpursley @ardaguclu How should we proceed with this? waiting for further discussion.